internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -362,14 +370,34 @@ func (r *BMACReconciler) handleBMHFinalizer(ctx context.Context, log logrus.Fiel
 	}
 
 	if bmhHasFinalizer {
+		log.Info("Handling BMH delete")
 		// agent could be nil here if it wasn't created yet, or if we deleted it, then failed to remove the finalizer for some reason
 		if agent != nil {
+			if result := r.configureBMHForCleaning(ctx, log, bmh); result.Stop(ctx) {
+				if _, err := result.Result(); err != nil {
+					log.WithError(err).Error("failed to reconfigure BMH for cleaning")
+				}
+				return result
+			}
+
+			if _, ok := agent.GetAnnotations()[BMH_FINALIZER_NAME]; !ok {
+				if agent.Annotations == nil {
+					agent.Annotations = make(map[string]string)
+				}
+				agent.Annotations[BMH_FINALIZER_NAME] = "true"
```

Comment:
What's the purpose of this annotation?

Reply:
if i understand correctly it's a mark for the agent to wait until bmh is deleted first
so it will mark with an annotation that the agent will monitor, should probably add a comment explaining a bit why the annotation is added, otherwise reading the code will need to go to the commit message.

Reply:
Right, this tells the agent to delete the spoke node when the BMH finishes deprovisioning.

I'll add a comment.